### ISSUE TITLE
Allow reset for some query parts

### ISSUE THREAD

**Issue by golgote** [auraphp/Aura.SqlQuery#84]
What about:
public function orderBy(array $spec, $replace = false)
instead of : public function orderBy(array $spec)

This can be added to other clauses as well.
I personally need this because my query is built by different objects and each adds some clauses that might later need to be replaced by others.

**Comment by harikt** [comment]
How about send a PR and in case @pmjones finds the time he can merge if that is ok :-) .

**Comment by golgote** [comment]
OK I will do that ASAP. I was just checking if you were open to this change.
Thanks.

**Comment by yespire** [comment]
Which syntax is better:
1. 
single reset() method handle all cases
query->reset()   // -- all, paramter is mixed optional
query->reset('order_by') // -- single factor
query->reset(['order_by', 'cols']  // -- multi-factors

2.
a set of specialized reset methods
query->reset() // -- all, no parameter
query->resetCols() // -- no parameter
query->resetOrderBy()

3.
support both 1 + 2

**Comment by harikt** [comment]
PR #91  .

**Comment by pmjones** [comment]
There are now 2 PRs in for this issue: #91 and #94. I will pick one of them, or implement the functionality myself. Thanks for your patience!

**Comment by pmjones** [comment]
@golgote @yespire @gauthier @harikt --

I am considering a series of `without*()` methods that return a clone of the current query. E.g.:

``` php
$newQuery = $oldQuery->withoutWhere()->withoutCols()->withoutOrderBy();
```

That way the existing query continues to exist, and the new one can be independently modified.

Thoughts?

**Comment by pmjones** [comment]
No, on further consideration, we already have some `reset*()` methods that can be exposed without BC break. I'll expand on those instead. If the user wants a clone, they can clone.

**Comment by golgote** [comment]
I like this idea, but sometimes one might just need to modify the original query, not create a new clone.

**Comment by harikt** [comment]
@pmjones FYI in case if you missed the discussion on https://github.com/auraphp/Aura.SqlQuery/pull/91#issuecomment-217611396 .

**Comment by pmjones** [comment]
Addressed by #95 et al. Let me know if those do not server your purposes.
